src/server/internalServer.cpp

```diff
@@ -451,28 +453,49 @@ bool InternalServer::start() {
   if (m_verbose.load())
     flags |= MHD_USE_DEBUG;
 
-  struct sockaddr_in sockAddr;
-  memset(&sockAddr, 0, sizeof(sockAddr));
-  sockAddr.sin_family = AF_INET;
-  sockAddr.sin_port = htons(m_port);
-  if (m_addr.empty()) {
-    if (0 != INADDR_ANY) {
-      sockAddr.sin_addr.s_addr = htonl(INADDR_ANY);
-    }
-    m_addr = kiwix::getBestPublicIp();
+
+  struct sockaddr_in sockAddr4={0};
+  struct sockaddr_in6 sockAddr6={0};
+
+
+  if (m_ipv6) {
+    flags|=MHD_USE_DUAL_STACK;
+    sockAddr6.sin6_family = AF_INET6;
+    sockAddr6.sin6_port = htons(m_port);
+    if (m_addr.empty()) {
+      if (0 != INADDR_ANY) {
+        sockAddr6.sin6_addr = in6addr_any;
+      }
+      m_addr = (kiwix::getBestPublicIp(m_ipv6)).addr6;
```

Comment:
Well, it looks like this is why we may need the `ipv6` boolean parameter. But are there realistic situations where the user trusts the application to select the IP address on its own but enforces the type of the IP address?

Reply:
None that I can think of.

One more case where the flag is needed is if a user does not specify an ip address but wants the server to support ipv6.